server/src/main/java/org/opensearch/search/query/QuerySearchResult.java

```diff
@@ -364,6 +366,11 @@ public void readFromWithId(ShardSearchContextId id, StreamInput in) throws IOExc
         nodeQueueSize = in.readInt();
         setShardSearchRequest(in.readOptionalWriteable(ShardSearchRequest::new));
         setRescoreDocIds(new RescoreDocIds(in));
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            tookTimeNanos = in.readVLong();
```

Comment:
Better to make this OptionalLong? ````in.readOptionalLong() ``` and set below as null instead of -1. 
Setting it -1 might be error prone if used and requires special checking. Though same for null but makes it more clear.